odap-package/src/odap/feature_factory/feature_store.py

```
@@ -60,3 +66,31 @@ def write_latest_table(latest_features_df: DataFrame, latest_table_name: str, la
 
     (latest_features_df.write.mode("overwrite").options(**options).saveAsTable(latest_table_name))
     logger.info("Write successful.")
+
+
+def generate_feature_lookups(
+    entity_name: str, features: Optional[List[str]], categories: List[str]
+) -> List[FeatureLookup]:
+    spark = SparkSession.getActiveSession()
+    config = get_config_namespace(ConfigNamespace.FEATURE_FACTORY)
+
+    entity_primary_key = get_entity_primary_key_by_name(entity_name, config)
+
+    metadata_table = get_metadata_table_for_entity(entity_name, config)
+    metadata = spark.read.table(metadata_table)
+
+    table_to_features_agg = (
+        metadata.filter(metadata.feature.isin(features) | metadata.category.isin(categories))
```

what if I enter a feature name or category which doesn't exist?

---

Well depends if we want to raise an exception if it doesn't exist

---

I think so, this would be failing silently otherwise
I would just make a diff between the inputs and the outputs and if not empty raise an exception showing the diff